internal/ethapi/api.go

```diff
@@ -549,9 +550,82 @@ type AccountResult struct {
 	StorageProof []StorageResult `json:"storageProof"`
 }
 type StorageResult struct {
-	Key   string       `json:"key"`
-	Value *hexutil.Big `json:"value"`
-	Proof []string     `json:"proof"`
+	Key     string       `json:"key"`
+	Value   *hexutil.Big `json:"value"`
+	Proof   []string     `json:"proof"`
+	Mutated bool         `json:"mutated"`
+}
+
+// Result structs for GetStateDiffProof
+type StateDiffProof struct {
+	Header   *HeaderMeta     `json:"header"`
+	Accounts []AccountResult `json:"accounts"`
+}
+type HeaderMeta struct {
+	Number    *big.Int    `json:"number"`
+	Hash      common.Hash `json:"hash"`
+	StateRoot common.Hash `json:"stateRoot"`
+	Timestamp uint64      `json:"timestamp"`
+}
+
+func (s *PublicBlockChainAPI) GetStateDiff(ctx context.Context, blockNrOrHash rpc.BlockNumberOrHash) (diffdb.Diff, error) {
+	_, header, err := s.b.StateAndHeaderByNumberOrHash(ctx, blockNrOrHash)
+	if err != nil {
+		return nil, err
+	}
+	return s.b.GetDiff(header.Number)
+}
+
+// GetStateDiffProof returns the Merkle-proofs corresponding to all the accounts and
+// storage slots which were touched for a given block number or hash.
+func (s *PublicBlockChainAPI) GetStateDiffProof(ctx context.Context, blockNrOrHash rpc.BlockNumberOrHash) (*StateDiffProof, error) {
```

Comment:
We did add a new RPC namespace `rollup`, I believe the `PublicBlockChainAPI` will expose in the `eth` namespace so it'll be `eth_getStateDiffProof`.  Would users expect this to be in the `eth` namespace or `rollup` namespace? See: https://github.com/ethereum-optimism/go-ethereum/blob/08903b8ee26084ef61e98bc5015dc2e20236b844/internal/ethapi/api.go#L1684

Reply:
No strong preference, I think either work. I decided to add it to eth's namespace to be next to GetProof